src/API/Endpoints/Reports/Endpoint.php

```diff
@@ -410,13 +410,70 @@ public function getPayments( $startStr, $endStr, $orderBy = 'date', $number = -
 		$payments = new \Give_Payments_Query( $args );
 		$payments = $payments->get_payments();
 
+		/**
+		 * Query renewal payments based on parent payments.
+		 * Because the renewal does not have a payment mode stored
+		 * the results are not included in the payment query,
+		 * so we have to run a second query and combine results.
+		*/
+		$payments = array_merge(
+			$payments,
+			$this->getRenewalPayments( $startStr, $endStr, $gateway )
+		);
+
 		// Cache the report data
 		$this->cachePayments( $args, $payments );
 
 		return $payments;
 
 	}
 
+	public function getRenewalPayments( $startStr, $endStr, $gateways ) {
+		global $wpdb;
+		$gateways          = implode(
+			',',
+			array_map(
+				function( $gateway ) use ( $wpdb ) {
+					return $wpdb->prepare( '%s', $gateway );
+				},
+				$gateways
+			)
+		);
+		$renewalPaymentIDs = $wpdb->get_col(
+			$wpdb->prepare(
+				"
+			SELECT renewals.ID
+			FROM {$wpdb->prefix}posts AS renewals
+			LEFT JOIN {$wpdb->prefix}give_donationmeta AS payment_mode
+				ON renewals.post_parent = payment_mode.donation_id
+			LEFT JOIN {$wpdb->prefix}give_donationmeta as payment_currency
+				ON renewals.ID = payment_currency.donation_id
+			LEFT JOIN {$wpdb->prefix}give_donationmeta as payment_gateway
+				ON renewals.ID = payment_gateway.donation_id
+			WHERE renewals.post_type = 'give_payment'
+			  AND renewals.post_parent != 0
+			  AND ( renewals.post_date BETWEEN '%s' AND '%s' )
+			  AND payment_mode.meta_key = '_give_payment_mode'
+			  AND payment_mode.meta_value = '%s'
+			  AND payment_currency.meta_key = '_give_payment_currency'
+			  AND payment_currency.meta_value = '%s'
+			  AND payment_gateway.meta_key = '_give_payment_gateway'
+			  AND payment_gateway.meta_value IN ($gateways)
+		",
+				$startStr,
+				$endStr,
+				( $this->testMode ) ? 'test' : 'live',
+				$this->currency
+			)
+		);
+		return array_map(
+			function( $renewalPaymentID ) {
+				return new Give_Payment( $renewalPaymentID );
+			},
+			$renewalPaymentIDs
+		);
```

Comment:
I believe this is going to cause an N + 1 problem wherein you are querying all of the payment IDs together, but then querying each payment object individually, therefore for every ID you have 1 additional query. Couldn't you instead query `renewals.*` and then generate the payment (or at least the `WP_Post`) from the data? That way it's all done in a single query.

Reply:
Ah, yup. I see the nested `get_post()` inside the `Give_Payment` definition.

Reply:
@JasonTheAdams the query is quickly getting out of hand, given all the granular data points that the reporting page uses for each payment. I may need to change course and query all donationmeta and then sort in PHP, as opposed to trying to join each of these different meta items in a single query.